### PR TITLE
make creation of http server by polka optional

### PR DESCRIPTION
**Pull request by ThiccCat**
Polka would create floating unused http server if people decided to use own like https or native written in C or other langs
by adding `{ createServer: true/false (default true) }` option it allows to control creation of internal server

I wasn't sure what to include in start method promise rejection if server doesn't exist so I leave it for you to decide

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -30,7 +32,7 @@ class Polka extends Router {
 
 	start(port, hostname) {
 		return new Promise((res, rej) => {
-			this.server.listen(port, hostname, err => err ? rej(err) : res());
+			this.server ? this.server.listen(port, hostname, err => err ? rej(err) : res()) : rej();
```
Comment (ThiccCat) [inline lib/index.js]: I am thinking of `rej('internal server disabled by options')`

**Comment by lukeed** [comment]
Hey, thanks! Will take some time to review as I figure out where I want Polka to go.

For the record though, re: https://github.com/lukeed/polka/issues/12#issuecomment-359166629, you can already wrap Polka with your own HTTP(S) server. The underlying server is only used/running if you call `start/listen` on Polka directly.

Please see the [`with-https` example](https://github.com/lukeed/polka/tree/master/examples/with-https) if you haven't already. Thanks!

**Comment by ThiccCat** [comment]
I acknowledged that it only listens if start is called but it still creates http server each time new class is created under .server property and that floats without usage.
Also, unused http servers number increases as more sub-apps are added

**Comment by ThiccCat** [comment]
#

**Pull request closed without merging**